### PR TITLE
Refreshing the writing policies in Go tutorial.

### LINKED ISSUE #321: Refresh writing policies in Go
- [ ] 01-intro-go.md
- [ ] 02-scaffold.md
- [ ] 03-policy-settings.md
- [ ] 04-validation.md
- [ ] 05-e2e-tests.md
- [ ] 06-logging.md
- [ ] 07-automate.md
- [ ] 08-distribute.md
- [ ] 09-validation-with-queries.md

### PR DESCRIPTION
**Pull request by jhkrug**
Will close #321 when merged.

### CODE REVIEW

**Comment by jhkrug** [comment]
Hi all,

The first 9 parts are ready for review. The last on raw policies, not yet. So, it's still draft, but waiting for review.

The last part on raw policies does not yet feel like it's in a tutorial format. Lots, of code, but I need more guidance on working with it, what goes in what files, when to test, input test data, test results, etc.

File: docs/writing-policies/go/07-automate.md
Comment (jvanz) [inline docs/writing-policies/go/07-automate.md:24]: Actually, the go policy template already came with two files [`test.yml` ](https://github.com/kubewarden/go-policy-template/blob/main/.github/workflows/test.yml)and `release.yml`. There is no need for rename the files. Furthermore, they make use of our github [actions](https://github.com/kubewarden/github-actions/blob/main/.github/workflows/reusable-test-policy-go.yml) that runs the linters and call `make test` and `make e2e-tests`.

File: docs/writing-policies/go/07-automate.md
Comment (jvanz) [inline docs/writing-policies/go/07-automate.md:37]: This is still required. But in a different way. As the files uses our github actions, the policy author need to change the [`oci-target`](https://github.com/kubewarden/go-policy-template/blob/ffef5e548495d9688d694316a4566dcf9ec88bec/.github/workflows/release.yml#L28) input used by the action

**Review by jvanz — CHANGES_REQUESTED** [review]

File: docs/writing-policies/go/02-scaffold.md
Comment (jhkrug) [reply docs/writing-policies/go/02-scaffold.md:80]: Adjusted, thanks.

File: docs/writing-policies/go/07-automate.md
Comment (jhkrug) [reply docs/writing-policies/go/07-automate.md:24]: Adjusted, thanks.

File: docs/writing-policies/go/07-automate.md
Comment (jhkrug) [reply docs/writing-policies/go/07-automate.md:37]: Adjusted, thanks.

File: docs/writing-policies/go/07-automate.md
Comment (jhkrug) [reply docs/writing-policies/go/07-automate.md:36]: Adjusted, thanks.

File: docs/writing-policies/go/03-policy-settings.md
Comment (jhkrug) [reply docs/writing-policies/go/03-policy-settings.md:31]: So, that entire 'type Settings struct' should be removed?

Comment (viccuad) [reply docs/writing-policies/go/03-policy-settings.md:32]: No, only the DeniedNames field of the struct.

**Review by viccuad — APPROVED** [review]
LGTM, modulo José's point.

File: docs/writing-policies/go/02-scaffold.md
Comment (jvanz) [inline docs/writing-policies/go/02-scaffold.md:27]: Why changing this regex? It looks like a typo.

File: docs/writing-policies/go/03-policy-settings.md
Comment (jvanz) [reply docs/writing-policies/go/03-policy-settings.md:32]: > No, only the DeniedNames field of the struct.

Exactly. :)

File: docs/writing-policies/go/07-automate.md
Comment (jvanz) [inline docs/writing-policies/go/07-automate.md:36]:
```suggestion
To enable the job, adjust the `oci-target` input for the reusable workflow (`reusable-release-policy-go.yml`) called in `release.yml` file.
```

File: docs/writing-policies/go/07-automate.md
Comment (jvanz) [inline docs/writing-policies/go/07-automate.md:73]: I think we need to rework the example section. This is using the old approch. Using the current reusable workflows, we should need to change the input for the `kubewarden/github-actions/.github/workflows/reusable-release-policy-go.yml` workflow. Take a look on this example from the `safe-labels` policy:

```yaml 
on:
  push:
    branches:
    - main
    tags:
    - 'v*'

name: Release policy

jobs:

  test:
    name: run tests and linters
    uses: kubewarden/github-actions/.github/workflows/reusable-test-policy-go.yml@v3.1.11

  release:
    needs: test
    permissions:
      # Required to create GH releases
      contents: write
      # Required to push to GHCR
      packages: write
      # Required by cosign keyless signing
      id-token: write

    uses: kubewarden/github-actions/.github/workflows/reusable-release-policy-go.yml@v3.1.11
    with:
      oci-target: ghcr.io/${{ github.repository_owner }}/policies/safe-labels #note here. This is line that requires a  change
```
This yaml came from [here](https://github.com/kubewarden/safe-labels-policy/blob/main/.github/workflows/release.yml). I think we can just copy it. Or copy only the call of the `kubewarden/github-actions/.github/workflows/reusable-release-policy-go.yml` workflow

**Review by jvanz — CHANGES_REQUESTED** [review]

File: docs/writing-policies/go/02-scaffold.md
Comment (jhkrug) [reply docs/writing-policies/go/02-scaffold.md:27]: I didn't. What should it be?

File: docs/writing-policies/go/07-automate.md
Comment (jhkrug) [reply docs/writing-policies/go/07-automate.md:36]: Ok, incorporated. Thanks.

File: docs/writing-policies/go/03-policy-settings.md
Comment (jhkrug) [reply docs/writing-policies/go/03-policy-settings.md:32]: ok, thanks.

Comment (jhkrug) [reply docs/writing-policies/go/03-policy-settings.md:32]: Hmm, ok. If I try that then settings.go no longer compiles as a in line 23 of settings .go a function IsNameDenied is defined which references DeniedNames. If the entire function IsNameDenied is removed then it all compiles and tests OK. So, remove both DeniedNames from the strcuture and the function IsNameDenied?

File: docs/writing-policies/go/07-automate.md
Comment (jhkrug) [reply docs/writing-policies/go/07-automate.md:73]: So which? Further suggestions?

File: docs/writing-policies/go/03-policy-settings.md
Comment (viccuad) [reply docs/writing-policies/go/03-policy-settings.md:32]: The correct thing to do would be to remove all mentions to DeniedNames including the IsNameDenied function and the tests. But that is a mouthful for first users of the guide.
Maybe leave the DeniedNames field but add a comment there with `// this is not needed, can be removed`?

File: docs/writing-policies/go/07-automate.md
Comment (viccuad) [reply docs/writing-policies/go/07-automate.md:73]: I think I prefer the current approach that only pushes the policy to an OCI registry. Using the reusable workflow is opaque to newcomers, and we do things there that aren't actually explained (e.g: signing, artifacthub) and are explained in ./distributing-policies.md.

@jvanz wdyt?

File: docs/writing-policies/go/03-policy-settings.md
Comment (jhkrug) [reply docs/writing-policies/go/03-policy-settings.md]: Ok, the first suggestion is the simplest and best and I'll implement that along with a note explaining.

**Comment by jhkrug** [comment]
Testing asciidoc

. one
. two
. three

Is that a numbered list?

**Review by viccuad — APPROVED** [review]

File: docs/writing-policies/go/02-scaffold.md
Comment (jvanz) [reply docs/writing-policies/go/02-scaffold.md]: Never mind, there is no change. It just the syntax highlight that looks like a removal for me.

File: docs/writing-policies/go/07-automate.md
Comment (jvanz) [reply docs/writing-policies/go/07-automate.md]: Fair enough. I agree with @viccuad. Let's keep the current approach.

**Review by jvanz — APPROVED** [review]

**Pull request merged**
